tools/test/collect_coverage.sh

```diff
@@ -107,11 +108,17 @@ fi
 
 cd $ROOT
 
-# If LCOV_MERGER is not set, use the legacy, awful, C++-only method to convert
-# coverage files.
-# NB: This is here just so that we don't regress. Do not add support for new
-# coverage features here. Implement it instead properly in LcovMerger.
-if [[ "$COVERAGE_LEGACY_MODE" ]]; then
+USES_LLVM_COV=
+if stat --printf='' "${COVERAGE_DIR}"/*.profraw 2>/dev/null; then
+  USES_LLVM_COV=1
+fi
+
+if [[ "$USES_LLVM_COV" ]]; then
+  llvm-profdata-3.9 merge -output "${COVERAGE_OUTPUT_FILE}" "${COVERAGE_DIR}"/*.profraw
```

Comment:
My machine has llvm-profdata-3.8 and llvm-profdata-4.0 :) Quick googling around suggests that its only llvm-profdata on osx. To make it run correctly on osx with xcode installed I read we should invoke it via xcrun (xcrun llvm-profdata).

One solution that we can use is to generate this file similarly to how crosstool is generated and expect users to provide relevant env variables. Thoughts?